Tests/CommandsTests/TestToolTests.swift

```diff
@@ -30,8 +43,47 @@ final class TestToolTests: XCTestCase {
         XCTAssert(try execute(["--version"]).contains("Swift Package Manager"))
     }
 
+    func testSanitizeNone() throws {
+        fixture(name: "Miscellaneous/SanitizersTest") { path in
+            _ = try execute([], packagePath: path)
+        }
+    }
+
+    func testSanitizeThread() throws {
+        fixture(name: "Miscellaneous/SanitizersTest") { path in
+            let cmdline = ["--sanitize=thread"]
+            XCTAssertThrows(BadCodeException.executionFailure) {
+                try executeTrash(cmdline, packagePath: path)
+            }
+        }
+    }
+
+    // Verifies that giving several options still works.
+    func testSanitizeThreadAddress() throws {
+        fixture(name: "Miscellaneous/SanitizersTest") { path in
+            let cmdline = ["--sanitize=thread", "--sanitize=address"]
+            XCTAssertThrows(BadCodeException.executionFailure) {
+                try executeTrash(cmdline, packagePath: path)
+            }
+        }
+    }
+
+    // Verifies that giving several options still works.
+    func testSanitizeAddressThread() throws {
+        fixture(name: "Miscellaneous/SanitizersTest") { path in
+            let cmdline = ["--sanitize=address", "--sanitize=thread"]
+            XCTAssertThrows(BadCodeException.executionFailure) {
+                try executeTrash(cmdline, packagePath: path)
+            }
+        }
+    }
+
     static var allTests = [
         ("testUsage", testUsage),
         ("testVersion", testVersion),
+        ("testSanitizeNone", testSanitizeNone),
+        ("testSanitizeThread", testSanitizeThread),
+        ("testSanitizeThreadAddress", testSanitizeThreadAddress),
+        ("testSanitizeAddressThread", testSanitizeAddressThread),
```

Comment:
@aciidb0mb3r this is not a functional test, this is mostly an integration test. It checks that:
 * the libraries are loaded correctly during runtime,
 * you can have more than one sanitizer library loaded
 * order of libraries doesn't matter
 * ...for `run`, which is executed directly
 * and for `test`, which goes through xctest on at least one platform.

Reply:
I am not suggesting that we shouldn't have these tests. We should unit test to make sure that the command line is constructed properly. And since all sanitizers go through same code path, we can have one of these tests (maybe `testSanitizeThreadAddress`) and rest are not really required IMO. But, if you think they're really important, they can be added in the [swift-integration-tests](https://github.com/apple/swift-integration-tests) repo.

Reply:
Left ThreadAddress.